analyzers/tests/SonarAnalyzer.UnitTest/TestCases/DebuggerDisplayUsesExistingMembers.CSharp11.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System.Diagnostics;
+
+class RawStringLiterals
+{
+    int SomeProperty => 1;
+    int SomeField = 2;
+
+    [DebuggerDisplay("""{SomeProperty}""")] int ExistingMemberTripleQuotes => 1;
+    [DebuggerDisplay(""""{SomeField}"""")] int ExistingMemberQuadrupleQuotes => 1;
+    [DebuggerDisplay("""
+        Some text{SomeField}
+        """)] int ExistingMultiLine => 1;
+    [DebuggerDisplay($$"""""
+        Some text{SomeField}
+        """"")] int ExistingMultiLineInterpolated => 1;
+
+    [DebuggerDisplay("""{Nonexistent}""")] int NonexistentTripleQuotes => 1;      // Noncompliant
+    //               ^^^^^^^^^^^^^^^^^^^
+    [DebuggerDisplay(""""{Nonexistent}"""")] int NonexistentQuadrupleQuotes => 1; // Noncompliant
+    //               ^^^^^^^^^^^^^^^^^^^^^
+    [DebuggerDisplay("""
+        Some text{Nonexistent}
+        """)] int NonexistentMultiLine1 => 1;                                     // Noncompliant@-2^22#46
+    [DebuggerDisplay("""
+        Some text{Some
+        Property}
+        """)] int NonexistentMultiLine2 => 1;                                     // FN: the new line char make the expression within braces not a valid identifier
+    [DebuggerDisplay($$"""""
+        Some text{Nonexistent}
+        """"")] int NonexistentMultiLineInterpolated => 1;                        // FN: interpolated raw string literals strings not supported
```

Comment:
[education] why?

Reply:
The implementation currently supports only literals, i.e. values that can be calculated at compile time.

Therefore, only single-line strings, as well as multi-line strings defined with `@` and raw string literals (`"""..."""`) are supported, as they generate `StringLiteralExpressionSyntax` instances on the AST.

Interpolated strings, single line (`$"..."`) or Verbatim single-line or multi-line (`@$"..."`, `$@"..."`), on the other hand, as well as raw interpolated strings of any sort(`$$..$""""..."""""`), are not literals, so they can't be computed at compile time.

To help visualizing all possibilities (known to me):
```cs
class A
{
	[DebuggerDisplay(@"...")] int Supported1 => 1;
	[DebuggerDisplay(@"...")] int Supported2 => 1;
	[DebuggerDisplay("""...""")] int Supported3 => 1;

	[DebuggerDisplay($".{x}..")] int NonSupported1 => 1; 	
	[DebuggerDisplay(@$".{x}..")] int NonSupported2 => 1;
	[DebuggerDisplay($@".{x}..")] int NonSupported3 => 1;
	[DebuggerDisplay($$""""".{{x}}..""""")] int NonSupported4 => 1; 
}
```

![image](https://user-images.githubusercontent.com/2676742/220349354-aa956a67-e3de-4dd3-b8ca-221cb7229e78.png)

Reply:
An additional note on an interesting scenario, I am not happy I am not covering is constant interpolated strings from C# 10:
```cs
public class ConstantInterpolatedStrings
{
    [DebuggerDisplay($"{{{nameof(NotAProperty)}}}")] // FN: constant interpolated strings not supported
    public int SomeProperty => 1;

    public class NotAProperty { }
}
```

Syntactically they are `InterpolatedStringExpressionSyntax`, semantically they are literal expressions, because `nameof` and `const` can all be resolved at compile time.

Moreover, it's not unlikely in my opinion, that a C# 10 developer would take advantage of this feature, to make sure he doesn't make syntactical mistakes when writing the name of the property between `{...}` in the string format parameter of the `DebuggerDisplay` attribute.

On the other hand, if `nameof` is used, one can only get wrong if he references the wrong name (as in the scenario reported above), or if a constant with an invalid identifier in it, is used.

Reply:
thanks!